R-package/R/create-release-files.R

```diff
@@ -0,0 +1,411 @@
+    
+    dictionary <- dictionary_info[[1]]
+    dictionary_version <- dictionary_info[[2]]
+    
+    # Validate files sheet
+    files_to_make <-
+      validate_files_sheet(dictionary,
+                           dictionary_version)
+    
+    create_files(files_to_make,
+                 dictionary)
+    
+    # Download previous release dictionary
+    if (is.null(past_dictionary_path)) {
+      past_dictionary_path <-
+        odm_dictionary$tmp_dictionary_directory_past_release
+      osfr::osf_auth(OSF_TOKEN)
+      osfr::osf_retrieve_file(OSF_LINK_PAST_RELEASE) %>%
+        osfr::osf_download(path = past_dictionary_path)
+    }
+    # Validate dictionary version
+    past_dictionary_info <- validate_version(past_dictionary_path)
+    
+    past_dictionary <- past_dictionary_info[[1]]
+    past_dictionary_version <- past_dictionary_info[[2]]
+    
+    # Validate files sheet
+    files_to_remove <-
+      validate_files_sheet(past_dictionary,
+                           past_dictionary_version)
+    
+    remove_files(files_to_make,
+                 dictionary)
+  }
+
+#' Validate Version
+#'
+#' Validate dictionary version between file name and summary sheets.
+#'
+#' @param dictionary_path string containing path to the dictionary directory.
+#'
+#' @return list with: Dictionary file name, Dictionary version, Boolean representing if the versions match.
+validate_version <- function(dictionary_path) {
+  # Acquire version number from file name
+  dictionary_version_pattern <- "ODM_dictionary_(\\d.*?).xlsx"
+  file_names <-
+    list.files(file.path(getwd(), dictionary_path),
+               pattern = dictionary_version_pattern)
+  # Display error and stop execution for multiple dictionaries
+  if (length(file_names) > 1) {
+    stop('Multiple dictionaries found, only one dictionary should be stored.')
+  } else if (length(file_names) == 0) {
+    stop("No valid files were detected. Make sure the dictionary file is named correctly.")
+  }
+  dictionary_file_name_version_number <-
+    regmatches(file_names,
+               regexec(dictionary_version_pattern, file_names))[[1]][2]
+  
+  # Read in the dictionary workbook
+  dictionary <- openxlsx::loadWorkbook(file.path(getwd(),
+                                                 dictionary_path,
+                                                 file_names))
+  
+  # Acquire version number from summary sheet
+  summary_sheet <-
+    openxlsx::readWorkbook(dictionary, odm_dictionary$summary_sheet_name)
+  # Read the first column
+  summary_versions <- summary_sheet[[1]]
+  # Strip off any NA rows
+  summary_versions <- summary_versions[!is.na(summary_versions)]
+  # Select the last version
+  summary_version <- summary_versions[[length(summary_versions)]]
+  # Compare if the versions match
+  if (summary_version != dictionary_file_name_version_number) {
+    logger::log_warn("Dictionary file name version does not reflect version in summary sheet")
```

Comment:
No since it could just be forgetting to change file name the rest could still be valid/benefit from being validated to notify the user of any additional issues.

Reply:
Continue execution, but stop the flow at a later stage once verification is complete to provide the user with the most feedback.